qiskit/circuit/instruction.py

```diff
@@ -188,11 +188,12 @@ def __eq__(self, other):
             return False
 
         for self_param, other_param in zip_longest(self.params, other.params):
-            try:
+            if isinstance(self_param, numpy.ndarray):
+                if numpy.array_equal(self_param, other_param):
+                    continue
+            else:
                 if self_param == other_param:
                     continue
-            except ValueError:
-                pass
```

Comment:
This _might_ have had an effect on timings, but since we gave standard gates their own `__eq__` overrides in #11370 it shouldn't make any difference at all (and when we move to Rust properly, it really won't).

Reply:
I was unsure about the performance implications of this. If the answer is none-in-the-future, I would prefer to avoid the `==` for ndarrays here.